libminifi/test/libtest/integration/IntegrationBase.cpp

```diff
@@ -28,9 +29,16 @@
 
 namespace org::apache::nifi::minifi::test {
 
-IntegrationBase::IntegrationBase(std::chrono::milliseconds waitTime)
+IntegrationBase::IntegrationBase(const std::optional<std::filesystem::path>& test_file_location, const std::optional<std::filesystem::path>& home_path, std::chrono::milliseconds waitTime)
     : configuration(std::make_shared<minifi::ConfigureImpl>()),
-      wait_time_(waitTime) {
+      wait_time_(waitTime),
+      home_path_(home_path) {
+  flow_config_path_.config_path = test_file_location;
+  if (test_file_location && std::filesystem::exists(*test_file_location) && std::filesystem::is_regular_file(*test_file_location)) {
```

Comment:
I think we can throw here if this condition is false; this should never happen.

Reply:
There are still some tests where no flow file path is used, also there are some tests where the test file is created later, so it may not be present yet when the harness is created.